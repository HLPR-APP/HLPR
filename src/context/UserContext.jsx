import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : null
  );
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const userCtx = useContext(UserContext);
  if (userCtx === undefined) {
    throw new Error('Must be wrapped in a UserProvider to use useUser');
  }
  return userCtx;
};

export { UserProvider, useUser };
