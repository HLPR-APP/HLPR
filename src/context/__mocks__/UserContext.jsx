import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : null
  );
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const userCtx = useContext(UserContext);
  if (userCtx === undefined) {
    throw new Error('Must be wrapped in a UserProvider to use useUser');
  }
  return userCtx;
};

export { UserProvider, useUser };
