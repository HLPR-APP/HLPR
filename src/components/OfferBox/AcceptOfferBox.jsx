import { Box, Badge, Image, Button } from '@chakra-ui/react';
import { Link } from 'react-router-dom';
import { updateOfferAccepted } from '../../services/services';
import { updateAcceptTask } from '../../services/services';
import { useUser } from '../../context/UserContext';

export default function AcceptOfferBox({ offer }) {
  const auth = useUser();
  const handleAcceptOffer = async (taskID, offerID) => {
    try {
      console.log(taskID);
      const response = await updateAcceptTask(auth.user.email, taskID);
      console.log(response);
      const returned = await updateOfferAccepted(offerID);
      console.log(returned);
      window.location.reload();
    } catch (err) {
      console.log(err.message);
    }
  };

  return (
    <>
      <Box
        maxW="sm"
        minW="sm"
        borderWidth="10px"
        borderRadius="lg"
        overflow="hidden"
        fit="cover"
      >
        <Badge borderRadius="full" px="2" colorScheme="teal" fontSize="md">
          {offer.Tasks2.name}
        </Badge>
        <Box p="6">
          <Box display="flex" alignItems="baseline">
            <Badge borderRadius="full" px="2" colorScheme="teal">
              OfferedBy:{<br />}
              {offer.offered_by}
            </Badge>
            <Box
              color="gray.500"
              fontWeight="semibold"
              letterSpacing="wide"
              fontSize="sm"
              textTransform="uppercase"
              ml="2"
            >
              Price: {offer.price}
            </Box>
          </Box>
          {!offer.accepted ? (
            <Button
              onClick={() => handleAcceptOffer(offer.Tasks2.id, offer.id)}
              size="md"
              margin="2"
              colorScheme="blue"
            >
              Accept Offer
            </Button>
          ) : (
            <Badge borderRadius="full" px="2" colorScheme="teal">
              Offer Accepted
            </Badge>
          )}
          <Box>
            <Box as="span" color="gray.600" fontSize="sm"></Box>
          </Box>
        </Box>
      </Box>
    </>
  );
}
