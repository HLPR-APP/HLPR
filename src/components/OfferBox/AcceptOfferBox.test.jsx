import dotenv from 'dotenv';
dotenv.config();
import React from 'react';
import { render } from '@testing-library/react';
import { UserProvider } from '../../context/UserContext.jsx';
import AcceptOfferBox from './AcceptOfferBox.jsx';
import { MemoryRouter } from 'react-router-dom';

jest.mock('../../services/users.js');
jest.mock('../../services/services.js');
jest.mock('../../context/UserContext.jsx');

const mockOffer = {
  Tasks2: {
    accepted_offer: null,
    created_at: '2022-01-19T04:24:08.238781',
    date: '2022-01-18',
    description: 'Somebody please do my laundry ASAP',
    id: 5,
    image_url:
      'https://images.unsplash.com/photo-1604335399105-a0c585fd81a1?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80',
    name: 'Folding Laundry',
    posted_by: 'stef45689@gmail.com',
  },
  accepted: false,
  date: null,
  duration: null,
  id: 37,
  offered_by: 'marvin_lambert@hotmail.com',
  price: 40,
  task_email: 'stef45689@gmail.com',
  task_id: 5,
  task_name: null,
};

it('should render offer box', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <AcceptOfferBox offer={mockOffer} />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
