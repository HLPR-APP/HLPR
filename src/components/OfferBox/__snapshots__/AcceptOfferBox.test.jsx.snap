// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render offer box 1`] = `
<div>
  <div
    class="css-bcjp8t"
  >
    <div
      class="css-60mfge"
      fit="cover"
    >
      <span
        class="chakra-badge css-gzm2p0"
      >
        Folding Laundry
      </span>
      <div
        class="css-1vae7t2"
      >
        <div
          class="css-x1sij0"
        >
          <span
            class="chakra-badge css-1de82h4"
          >
            Offered By:
            <br />
            marvin_lambert@hotmail.com
          </span>
          <div
            class="css-10cn095"
          >
            Price: $
            40
          </div>
        </div>
        <div
          class="css-gmuwbf"
        >
          <button
            class="chakra-button css-1y3eg1"
            type="button"
          >
            Accept Offer
          </button>
        </div>
        <div
          class="css-0"
        >
          <span
            class="css-bgvdb6"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
